SDK/New Group/Models/SettingsObject.swift

```diff
@@ -0,0 +1,78 @@
+//
+//  SettingsObject.swift
+//  iOS-SDK
+//
+//  Created by Alberto Huerdo on 12/17/18.
+//  Copyright © 2018 SchedJoules. All rights reserved.
+//
+
+import Foundation
+import SchedJoulesApiClient
+
+struct SettingsObject: CodedOption {
+    
+    enum CodingKeys: String, CodingKey {
+        case code = "code"
+        case name = "name"
+        case path = "path"
+    }
+    
+    
+    var name: String
+    var code: String
+    var icon: URL?
+    
+    var path: String?
+    
+    init(name: String, code: String, iconURL: URL?) {
+        self.name = name
+        self.code = code
+        self.icon = iconURL
+    }
+    
+    init(object: CodedOption, type: SettingsManager.SettingsType) {
+        self.name = object.name
```

Comment:
Please make sure there is only/exactly one "primary" (or "required") constructor and all the others just call that one. Only the primary ctor should assign and member variables. For instance, the 3rd constructor would look like this:

```Swift
init () {
    self.init(name: "Default", code: "Default", iconURL: nil)
}
```
I'm not entirely sure if that's valid Swift, but you get the idea.
The goal is to reduce/avoid code duplication. In general it's better to delegate to another class/method/ctor which already implements the behaviour than implementing the same code again.